src/terrain/terrain_helper.hpp

```diff
@@ -40,21 +40,22 @@ grow_grass_inner(Terrain& ter, std::set<Tile*> in_grass, int height) {
     // the next level down
     std::set<Tile*> next_grass_tiles;
     for (Tile* tile : in_grass) {
-        for (int x = -1; x < 2; x++)
-            for (int y = -1; y < 2; y++) {
-                Tile* adjacent_tile = Ninos_horrible_function(ter, tile, x, y);
-                if (adjacent_tile == nullptr){
-                    continue;
-                }
-                // instead of height used maxheight +1
-                if (adjacent_tile->is_grass() && (getter(adjacent_tile) < height)) {
-                    // if adjacent tile is on the level below
-                    // it should be added to the next iteration
-                    next_grass_tiles.insert(adjacent_tile);
-                    // and the tiles with height less should have their height set
-                    setter(adjacent_tile, height);
-                }
+        for (auto it = ter.get_tile_adjacent_iterator(
+                 ter.pos(tile),
+                 DirectionFlags::HORIZONTAL1 | DirectionFlags::HORIZONTAL2
+             );
+             !it.end(); it++) {
```

Comment:
Move the iterator outside:
```suggestion
        auto it = ter.get_tile_adjacent_iterator(
                 ter.pos(tile),
                 DirectionFlags::HORIZONTAL1 | DirectionFlags::HORIZONTAL2
        );
        for (; !it.end(); it++) {
```